doc: migration_guide: Add notes (#12750)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/migration/migration_guide_2.5.rst

    UART is now configured using only devicetree.
  * The :kconfig:option:`CONFIG_SLM_CARRIER_APP_DATA_CONTAINER_BUFFER_LEN` Kconfig option has been renamed to :kconfig:option:`CONFIG_SLM_CARRIER_APP_DATA_BUFFER_LEN`.

> * Applications using MCUboot must update their version from Kconfig ``CONFIG_MCUBOOT_IMAGE_VERSION`` to using an :ref:`application VERSION file <zephyr:app-version-details>` to set the version.

Reviewer: ```suggestion
* Applications using MCUboot must update to setting their version  using an :ref:`application VERSION file <zephyr:app-version-details>` instead of the previously used ``CONFIG_MCUBOOT_IMAGE_VERSION`` Kconfig option.
```

Author: Updated. Any reason as to why this part appears bold? https://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-12750/nrf/releases_and_maturity/migration/migration_guide_2.5.html

Reviewer: One too many spaces in the beginning of the second line, so the build system takes it as a term definition.

>   Alternatively, the version can be set in Kconfig by using the :kconfig:option:`CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION` Kconfig option, but using a :file:`VERSION` file is the recommended approach.

Reviewer 2: ```suggestion
  Alternatively, you can set the version in Kconfig by using the :kconfig:option:`CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION` option, but using a :file:`VERSION` file is the recommended approach.
```

Author: Fixed

>    Alternatively, the version can be set in Kconfig by using the :kconfig:option:`CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION` Kconfig option, but using a :file:`VERSION` file is the recommended approach.

Reviewer: ```suggestion
  Alternatively, the version can be set in Kconfig by using the :kconfig:option:`CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION` Kconfig option, but using a :file:`VERSION` file is the recommended approach.
```

Author: Fixed

  * If you are using the :ref:`liblwm2m_carrier_readme` library, make sure to take into account the addition of the auto-connect feature that is enabled by default.
  * When performing a modem firmware update, you can now reset only the modem (instead of the whole device) using the new ``#XMODEMRESET`` AT command.

> * Applications which use :file:`prj_<board>.conf` Kconfig configurations should be transitioned to using :file:`boards/<board>.conf` Kconfig fragments.

Reviewer 2: ```suggestion
* Applications that use :file:`prj_<board>.conf` Kconfig configurations should be transitioned to using :file:`boards/<board>.conf` Kconfig fragments.
```

Author: Fixed
